src/error.rs

```diff
@@ -12,9 +13,13 @@ use std::{fmt, str};
 /// An Error is also returned in case ALSA returns a string that
 /// cannot be translated into Rust's UTF-8 strings.
 #[derive(Clone, PartialEq, Copy)]
-pub struct Error(&'static str, i32);
+pub struct Error(&'static str, c_int);
 
-pub type Result<T> = ::std::result::Result<T, Error>;
+pub type Result<T> = ::core::result::Result<T, Error>;
+
+extern "C" {
```

Comment:
I think we should have different implementations for std and no-std mode. So that the errno import is only done in no-std mode, keeps the std mode cleaner. Like

```
#feature(std)
std::io::Error::last_os_error().raw_os_error().unwrap_or_default()
#feature(no-std)
unsafe { errno }
```